packages/gatsby/cache-dir/app.js

```diff
@@ -110,11 +126,15 @@ apiRunnerAsync(`onClientEntry`).then(() => {
     loader.loadPage(window.location.pathname),
   ]).then(() => {
     const preferDefault = m => (m && m.default) || m
-    let Root = preferDefault(require(`./root`))
+    const Root = preferDefault(require(`./root`))
     domReady(() => {
-      renderer(<Root />, rootElement, () => {
-        apiRunner(`onInitialClientRender`)
-      })
+      // Hot reloading can by-pass the above loadPage promise so we
+      // run this again.
+      loader.loadPage(window.location.pathname).then(() =>
```

Comment:
Can you elaborate a bit on how it can bypass the `loadPage` promise? Not entirely clear from this comment.

Reply:
not entirely sure about this now — lemme double-check

Reply:
fake news. Removed.